### DISCUSSION TITLE
Environment variables

### DISCUSSION THREAD

**Discussion by bwlove319** [theory/sqitch#612] [Q&A]
Why can't sqitch read environment variables?

Snowsql can... but sqitch apparently can't.
There should be:
SQITCH_PWD
SQITCH_USER
... etc...

It appears the ONLY way to give sqitch a Snowflake password is by specifying it in the URI, which means my clients must either:
- store the password in the sqitch.conf file (not acceptable), or
- know the format of the URI and type it every time (also not acceptable)

The inability to read configs from env vars makes using Sqitch exponentially more difficult.

In fact, there is no place in the docs that definitively states exactly what configs need to be set and where.
Apparently, some go in the .snowsql/config, some go in odbc.ini, and some go in sqitch.conf, but even then - with a database specified in all three places - it tells me no database is specified.

Very confusing.  Just let it read environment variables.

**Comment by theory** [comment]
Sqitch supports most of the Snowflake environment variables. Have a look at [sqitch-environment](http://sqitch.org/docs/manual/sqitch-environment/) for details.

**Comment by bwlove319** [comment]
Ok, I see that now...
I was looking in `Setup and Config` for that info... the link to the env vars page was down below, out of sight.
I was also expecting to see references to the env vars in either:
- the help message for Sqitch, or
- the config docs

Like how Snowflake (et al) does it:
![image](https://user-images.githubusercontent.com/87549506/156627104-d550d28e-24d6-494d-9b00-995776dbda5f.png)

![image](https://user-images.githubusercontent.com/87549506/156627220-0f08e9ca-3e70-41c3-82c6-f7516ce7d8cb.png)

I think it would help a lot to reference those env vars in more predictable places, or at least add a link to them in the `Setup and Config` section.  I did page search for "env" and found nothing in Setup and Config.

**Reply by theory** [reply]
Sqitch supports [a lot of environment variables](http://sqitch.org/docs/manual/sqitch-environment/), different ones for each engine. They would don't all fit in the usage statement, and it'd would be confusing if they did.
